Fix and test for issue #194

qqilihq opened the pull request:
Fix for error handling for `fileSize` limit; in case diskStorage was used, route was called when limit was hit, instead of error.

LinusU commented:
Would you mind fixing the following style issues so that the travis build will run :+1: 

```text
standard: Use JavaScript Standard Style (https://github.com/feross/standard)
  test/express-integration.js:18:17: Missing space before function parentheses.
  test/express-integration.js:19:15: Expected error to be handled.
  test/express-integration.js:19:23: Missing space before function parentheses.
  test/express-integration.js:21:13: Extra semicolon.
  test/express-integration.js:22:7: Extra semicolon.
  test/express-integration.js:24:16: Missing space before function parentheses.
  test/express-integration.js:30:26: Trailing spaces not allowed.
  test/express-integration.js:31:22: Trailing spaces not allowed.
  test/express-integration.js:32:47: Missing space before value for key "destination".
  test/express-integration.js:32:59: Trailing spaces not allowed.
```

LinusU commented on test/express-integration.js:
```diff
@@ -8,14 +8,29 @@ var util = require('./_util')
+    });
+  })
+  after(function() {
+    rimraf.sync(uploadDir)
```
Not that important but I would prefer:

```javascript
after(function (done) {
  rimraf(uploadDir, done)
```

qqilihq commented:
Sorry, fixed now.

LinusU commented:
No problem, I'm trying to figure out why this works. I remember that I had the `abort` in an upper scope, such as you have now, but moved it into a tighter scope. I don't remember why thought. Would you mind explaining the code?

Also, maybe the check for aborting should be moved to `indicateDone` instead of in `multer.on('finish')`...

qqilihq commented:
I moved the checking of the aborting flag, as suggested.

My problem was, that in case I was using the disk storage, and the limit was hit, the express route handler was getting called, instead of the error handler. That's why I introduced the `!aborting` check within the indicateDone (in case, the upload is canceled because of file size limit, the `done(uploadError)` callback (line 54) will be triggered.

Not sure why the CI test is failing now, works locally for me?

LinusU commented:
I restarted the CI job, it seems like it worked this time :confused:

qqilihq commented:
Hi Linus, not sure about the reason for the fail above. Running it locally several times gave no failure at all.

LinusU commented:
This should be fixed by #205 now :)

qqilihq commented:
Hi Linus, thank you, however I'm still having that problem. As soon as I use the `diskStorage`, the error event is not called when exceeding `fileSizeLimit`. In case of using the default storage, everything is fine. Test case demonstrating the issue is my first commit above.

Best,
Philipp
